modules/runtime-info/src/main/java/org/opencastproject/runtimeinfo/RuntimeInfo.java

```diff
@@ -218,6 +233,149 @@ public String getMyInfo() {
     return gson.toJson(result);
   }
 
+  @GET
+  @Path("health")
+  @Produces("application/health+json")
+  @RestQuery(name = "health", description = "Opencast node health check. Implements this internet-draft health check api https://inadarei.github.io/rfc-healthcheck",
+          reponses = {
+            @RestResponse(responseCode = HttpServletResponse.SC_OK, description = "Node is running, check reponse for details"),
+            @RestResponse(responseCode = HttpServletResponse.SC_SERVICE_UNAVAILABLE, description = "Node is offline or unresponsive, check response for details")},
+          returnDescription = "Details of the Opencast node's health status")
+
+  public String getHealth(@Context HttpServletResponse response) {
+    /* Response implements https://inadarei.github.io/rfc-healthcheck
+     * Example reponse
+    {
+        "description" : "Opencast node's health status",
+        "releaseId" : "TEST",
+        "checks" : {
+           "service:states" : [
+              {
+                 "observedValue" : "WARNING",
+                 "links" : {
+                    "path" : "service1"
+                 },
+                 "changed" : "Tue Jun 04 11:15:27 BST 2019",
+                 "componentId" : "service1"
+              },
+              {
+                 "changed" : "Tue Jun 04 11:15:27 BST 2019",
+                 "links" : {
+                    "path" : "service2"
+                 },
+                 "observedValue" : "ERROR",
+                 "componentId" : "service2"
+              }
+           ]
+        },
+        "notes" : [
+           "service(s) in WARN state",
+           "service(s) in ERROR state"
+        ],
+        "status" : "warn",
+        "serviceId" : "http://localhost",
+        "version" : "1"
+    }
+     */
+    String status = HEALTH_CHECK_STATUS_PASS; // pass, warn or fail
+    // Conditional workaround for unit tests
+    String releaseId = this.bundleContext != null ? this.bundleContext.getBundle().getVersion().toString() : "TEST";
+    String hostname = serviceRegistry.getRegistryHostname();
+    Gson gson = new Gson();
+
+    List<String> notes = new ArrayList<>();
+    List<Object> serviceStates = new ArrayList<>();
+    Map<String, Object> checks = new HashMap<>();
+
+    try {
+      HostRegistration host = serviceRegistry.getHostRegistration(hostname);
+
+      // check most severe conditions first
+      if (!host.isOnline()) {
+        // NOTE: This is not strictly possible as a node can't test if it's offline
```

Comment:
I could see the case for extending this endpoint to be able to check on the health of *other* nodes from the current one...

Reply:
As this is primarily for use by load balancers, that would not make sense initially. Maybe an option to check the cluster or separate endpoints e.g. /info/health/all,  /info/health/<host>

Reply:
Yep exactly.  Future work, not really relevant here, but it's better to have the logic present so that it doesn't cause issues down the line :)

Reply:
Without changing anything we could simply aggregate the responses from all the nodes by calling /info/health for each one.

Alternative just adding an additional checks object for "node:statuses" and use the observedValue to return pass|warn|fail. This can be added in the future without breaking the format.

For now I'll just look at breaking  the host analysis out into a separate function